gattc_sd.go

```diff
@@ -17,6 +17,11 @@ import (
 	"runtime/volatile"
 )
 
+const (
+	maxNumberServicesToDiscover        = 8
+	maxNumberCharacteristicsToDiscover = 16
```

Comment:
Strictly speaking this is not a maximum. Also, I'm fine with `append` when the number of services/characteristics is not known in advance (there is no way to get the allocation right). In fact, I think these numbers could be lowered a bit to more likely numbers of services/characteristics, such as maybe 4-6 services and maybe 4-8 characteristics. Using `append` makes sure more memory will be allocated as needed.

(not that important, not blocking a merge)